g11n-ws/modules/md-data-api-s3impl/src/main/java/com/vmware/vip/messages/data/dao/impl/S3MultComponentDaoImpl.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright 2019 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.vmware.vip.messages.data.dao.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import com.vmware.vip.common.constants.ConstantsChar;
+import com.vmware.vip.messages.data.dao.api.IMultComponentDao;
+import com.vmware.vip.messages.data.dao.api.IOneComponentDao;
+import com.vmware.vip.messages.data.dao.exception.DataException;
+import com.vmware.vip.messages.data.dao.model.ResultI18Message;
+/**
+ * this class get the bundle json from bundle file
+ */
+@Repository
+public class S3MultComponentDaoImpl implements IMultComponentDao {
+   private static Logger logger = LoggerFactory.getLogger(S3MultComponentDaoImpl.class);
+
+   @Autowired
+   private IOneComponentDao oneComponentDao;
+
+   /**
+    * get the bundle files from s3 server
+    */
+   @Override
+   public List<String> get2JsonStrs(String productName, String version, List<String> components,
+         List<String> locales) throws DataException {
+      logger.debug("begin get2JsonStrs");
+      List<String> bundles = new ArrayList<>();
+      if (components == null || locales == null) {
+         throw new DataException("S3 No component or locale");
```

Comment:
Define a common const to store error message.  
"Doesn't provide component or locale!"